### PR TITLE
adding new apis for mm relayer

### PR DESCRIPTION
**Pull request by zhangqi021**

### CODE REVIEW

File: sdk/proto/service/rfqmm/api.proto [hunk 1]
```diff
@@ -20,6 +20,64 @@ service api {
       body: "*"
     };
   };
+  rpc SignQuoteHash(SignQuoteHashRequest) returns (SignQuoteHashResponse) {
+    option (google.api.http) = {
+      post: "/v1/rfqmm/signQuoteHash"
+      body: "*"
+    };
+  };
+  rpc Sign(SignRequest) returns (SignResponse) {
```
Comment (ClaudeZsb) [inline sdk/proto/service/rfqmm/api.proto:29]: What does this `Sign` used for? Same question for `Verify` rpc

File: sdk/service/rfqmm/service.go [hunk 2]
```diff
@@ -61,6 +61,11 @@ type ServerConfig struct {
 	TPPolicyList []string
 	// port num that mm would listen on
 	PortListenOn int64
+	// light mm, which needs a relayer to interact with rfq server
+	LightMM bool
+	// if not set, will use localhost
+	Host string
+	Port string
```
Comment (ClaudeZsb) [inline sdk/service/rfqmm/service.go:68]: Is it for replacing `PortListenOn`? Seems like `Port` is not used.

File: sdk/service/rfqmm/api.go [hunk 3]
```diff
@@ -29,13 +32,13 @@ func (c *Client) Quote(ctx context.Context, in *proto.QuoteRequest, opts ...grpc
 
 func (s *Server) Price(ctx context.Context, request *proto.PriceRequest) (response *proto.PriceResponse, err error) {
 	// uncomment it out for easy debugging
```
Comment (ClaudeZsb) [inline sdk/service/rfqmm/api.go:34]: remove this comment.

File: sdk/service/rfqmm/api.go [hunk 4]
```diff
@@ -143,6 +140,97 @@ func (s *Server) Quote(ctx context.Context, request *proto.QuoteRequest) (respon
...
+}
+
+func (s *Server) SignQuoteHash(ctx context.Context, request *proto.SignQuoteHashRequest) (*proto.SignQuoteHashResponse, error) {
+	dstChainId := request.GetQuote().GetDstChainId()
+	rfqContract, err := s.ChainCaller.GetRfqContract(dstChainId)
+	if err != nil {
+		return signQuoteHashArgumentErr(err.Error())
+	}
+
+	// 0. check quote hash
```
Comment (ClaudeZsb) [inline sdk/service/rfqmm/api.go:162]: `check quote sig`

File: sdk/service/rfqmm/api.go [hunk 5]
```diff
@@ -143,6 +140,97 @@ func (s *Server) Quote(ctx context.Context, request *proto.QuoteRequest) (respon
...
+		return &proto.SignResponse{
+			Err: err.(*proto.Err).ToCommonErr(),
+		}, nil
+	}
+	return &proto.SignResponse{
+		Sig: sig,
+	}, nil
+}
+
+func (s *Server) SignQuoteHash(ctx context.Context, request *proto.SignQuoteHashRequest) (*proto.SignQuoteHashResponse, error) {
```
Comment (ClaudeZsb) [inline sdk/service/rfqmm/api.go:155]: The biggest problem is that if an mm is not light-versioned, this api is still available. And that might damage the mm's fund security. Might need to check mm's version before signing any data.

File: sdk/service/rfqmm/service.go (hunk 2 above)
Comment (zhangqi021) [reply sdk/service/rfqmm/service.go:68]: removed

File: sdk/proto/service/rfqmm/api.proto (hunk 1 above)
Comment (zhangqi021) [reply sdk/proto/service/rfqmm/api.proto:29]: not used now. removed

File: sdk/service/rfqmm/api.go (hunk 5 above)
Comment (zhangqi021) [reply sdk/service/rfqmm/api.go:142]: good suggestion, done

**Review by ClaudeZsb — APPROVED** [review]

File: sdk/service/rfqmm/api.go [hunk 6]
```diff
@@ -143,6 +139,82 @@ func (s *Server) Quote(ctx context.Context, request *proto.QuoteRequest) (respon
 	return &proto.QuoteResponse{QuoteSig: eth.Bytes2Hex(sigBytes)}, nil
 }
 
+func (s *Server) SignQuoteHash(ctx context.Context, request *proto.SignQuoteHashRequest) (*proto.SignQuoteHashResponse, error) {
```
Comment (xiaozhou) [inline sdk/service/rfqmm/api.go:142]: too many duplicated check logics with server.processOrder(), can we merge these logics and let them use a same verify function?

Comment (zhangqi021) [reply sdk/service/rfqmm/api.go:140]: verify logic merged

**Review by xiaozhou — APPROVED** [review]

**Pull request merged**